src/protagonist/API/Features/Image/Requests/CreateOrUpdateImage.cs

```diff
@@ -114,6 +114,11 @@ public async Task<ModifyEntityResult<Asset>> Handle(CreateOrUpdateImage request,
 
         if (modifyEntityResult.IsSuccess)
         {
+            if (modifyEntityResult.Entity!.HasSingleDeliveryChannel(AssetDeliveryChannels.Timebased))
```

Comment:
I don't think this is correct, if an asset has both `["iiif-av", "file"]` it would fail this test but we'd still want to create a batch asset. 

This logic also deviates from when the batch is created - is it simpler to keep them together on the same code path? That way we would ensure that if a batch is created, we'd always create a `BatchAsset` (and conversely never create a `BatchAsset` if we don't create a `Batch`)?